lib/puppet/util/windows/registry.rb

```diff
@@ -65,6 +65,25 @@ def values(key)
       vals
     end
 
+    # Retrieve a set of values from the registry for a given key and list of value names
+    # Value names listed but not found in the registry will not be added to the resultant
+    # hashtable
+    #
+    # @param key [RegistryKey] An open handle to a Registry Key
+    # @param value_names [String[]] An array of keynames that should be returned if they exist
+    # @return [Hashtable<String, String>] A hashtable of all of the found values in the registry key
+    def values_by_name(key, value_names)
+      vals = {}
+      reg_enum_valuenames(key).each do |value_name|
```

Comment:
FYI - This changes an O(n) to an O(n^2).

Reply:
Not really true...Before it would enum and read all keys (because enum called read), whereas my way enumed all keys but only read those relevant.  Worst case same, performance, best case, slightly better.  So still O(n)

Reply:
Oh I see the O(n^2) now.... I could have probably yielded instead.  Performance wise I doubt you would've seen it though.